bin/demo.py

```diff
@@ -31,97 +33,75 @@
+    except:
+        log_and_exit('!! authentication failed')
 
 def config_dcos_cli(dcos_url):
-    if call (["dcos", "config", "set", "core.dcos_url", dcos_url],stdout=open(os.devnull, 'wb')) == 1:
-        log_and_exit ("Unable to configure DCOS CLI.")
-
-def make_temp_dir():
-    remove_temp_dir()
-    os.mkdir("tmp")
-
-def rename(path, jenkins_name):
-    with open(path, 'r+') as f:
-        config = json.load(f)
-        config['jenkins']['framework-name'] = jenkins_name
-        f.seek(0)
-        json.dump(config, f, indent=4)
-
-def install(dcos_url, jenkins_name, jenkins_url):
-    log ("Installing Jenkins with name {}.".format(jenkins_name))
-    shutil.copyfile("conf/jenkins.json", "tmp/jenkins.json")
-    rename("tmp/jenkins.json", jenkins_name)
-    command = "dcos package install --yes --options=tmp/jenkins.json jenkins"
-    print ("\n> " + command)
-    if call (['dcos', 'package', 'install', '--yes', '--options=tmp/jenkins.json', 'jenkins']) != 0:
-        log_and_exit ("Failed to install Jenkins.")
-    log("Jenkins has been installed! Wait for it to come up before proceeding at: {}".format(jenkins_url))
-
-def verify(jenkins_url):
-    r = requests.get(jenkins_url, headers=auth_func({}), verify=False)
-    if r.status_code != 200:
-        log ("Couldn't find a Jenkins instance running at {}.".format(jenkins_url))
+    dcos.config.set_val('core.dcos_url', dcos_url)
+    dcos.config.set_val('core.ssl_verify', 'False')
+
+def install_jenkins(jenkins_name, jenkins_url):
+    log("installing Jenkins with name '{}'".format(jenkins_name))
+    install_package_and_wait('jenkins', None, jenkins_name)
+    assert package_installed('jenkins'), log_and_exit('!! package failed to install')
+    log("waiting for Jenkins service to come up at '{}'".format(jenkins_url))
+    end_time = time.time() + 300
+    while time.time() < end_time:
+        if verify_jenkins(jenkins_url):
+            break
+        time.sleep(1)
+
+def verify_jenkins(jenkins_url):
+    try:
+        r = http.get(jenkins_url)
+        if r.status_code == 200 and r.headers['x-jenkins']:
+            log("service is up and running, got Jenkins version '{}'".format(r.headers['x-jenkins']))
+            return True
+    except:
         return False
-    log ("Jenkins is up and running! Got Jenkins version {}".format(r.headers['x-jenkins']))
-    return True
 
-def check_marathon_lb(elb_url):
-    log("Checking to see if Marathon-lb is installed.")
+def install_marathon_lb(marathon_lb_url):
+    log("installing marathon-lb")
+    install_package_and_wait('marathon-lb')
+    assert package_installed('marathon-lb'), log_and_exit('!! package failed to install')
+    log("waiting for marathon-lb service to come up at '{}'".format(marathon_lb_url))
+    end_time = time.time() + 300
+    while time.time() < end_time:
+        if verify_marathon_lb(marathon_lb_url):
+            break
+        time.sleep(1)
+
+def verify_marathon_lb(marathon_lb_url):
```

Comment:
Same question as the `verify_jenkins` method - does this return `False` on a `503`?